agents/craftassist/craftassist_agent.py

```diff
@@ -185,9 +185,15 @@ def init_controller(self):
         )
 
     def perceive(self, force=False):
+        """Whenever some blocks are changed, that area will be put into a 
+        buffer which will be force-perceived by the agent in the next step
+
+        Here the agent first clusters areas that are overlapping on the buffer,
+        then run through all perception modules to perceive
+        and finally clear the buffer when perception is done.
+        """
         self.areas_to_perceive = cluster_areas(self.areas_to_perceive)
-        for v in self.perception_modules.values():
-            v.perceive(force=force)
+        super().perceive()
```

Comment:
Ah this is a good catch. Tagging @anuragprat1k here to also take a look at `locobot_agent.py` to make sure we are calling `super.perceive()` there.

Reply:
I took a look at `locobot_agent.py` as well, seems that locobot isn't overwriting `perceive()` -- it's directly using `perceive()` from `loco_mc_agent.py`

Reply:
Thanks for taking a look!